src/java.rs

```diff
@@ -16,61 +16,46 @@ fn java_base_types(encoded_ty: char) -> Option<&'static str> {
 }
 
 fn byte_code_type_to_java_type(byte_code_type: &str, mapper: &ProguardMapper) -> Option<String> {
-    let mut chrs = byte_code_type.chars();
-    let token = chrs.next()?;
-    if token == 'L' {
-        // invalid signature
-        let l = chrs.clone().last();
-        if l.is_none() || l.unwrap() != ';' {
-            return None;
-        }
-        chrs.next_back(); // remove final `;`
-        let obfuscated = chrs.collect::<String>().replace('/', ".");
+    let mut chrs = byte_code_type.char_indices();
+    //let (idx, token) = chrs.next()?;
+    let mut suffix = String::new();
+    while let Some((idx, token)) = chrs.next() {
+        if token == 'L' {
+            // expect and remove final `;`
+            if chrs.next_back()?.1 != ';' {
+                return None;
+            }
+            let obfuscated = byte_code_type
+                .get(idx + 1..byte_code_type.len() - 1)?
+                .replace('/', ".");
 
-        if let Some(mapped) = mapper.remap_class(&obfuscated) {
-            return Some(mapped.to_string());
-        }
+            if let Some(mapped) = mapper.remap_class(&obfuscated) {
+                return Some(format!("{}{}", mapped, suffix));
+            }
 
-        return Some(obfuscated);
-    } else if token == '[' {
-        let type_sig = chrs.clone().collect::<String>();
-        if !type_sig.is_empty() {
-            return Some(format!(
-                "{}[]",
-                byte_code_type_to_java_type(chrs.collect::<String>().as_str(), mapper)
-                    .unwrap_or_default()
-            ));
+            return Some(format!("{}{}", obfuscated, suffix));
+        } else if token == '[' {
+            suffix.push_str("[]");
```

Comment:
```suggestion
            suffix = "[]";
```

Reply:
this suggestion is wrong, my bad. Forgot that you can have multiple `[`.

Reply:
here I need to use `suffix.push_str("[]");` as I might have more than one, for example: `[[[` so I need to be able to add more `[]` at each iteration.